doc/plugins.rst

```diff
@@ -5,74 +5,175 @@ tox plugins
+    you could apply `cookiecutter -f cookiecutter-tox-plugin` and adapt the code to the
+    package based structure used in the cookiecutter.
 
-.. code-block:: shell
+Let us consider you want to extend tox behaviour by displaying fireworks at the end of a
+successful tox run (we won't go into the details of how to display fireworks though).
 
-    python setup.py sdist register upload
+To create a working plugin you need at least a python project with a tox entry point and a python
+module implementing one or more of the pluggy based hooks tox specifies (using the
+``@tox.hookimpl`` decorator as marker).
 
+minimal structure:
+
+.. code-block:: shell
 
-Writing hook implementations
-----------------------------
+    $ mkdir tox-fireworks
+    $ cd tox-fireworks
+    $ touch tox_fireworks.py
+    $ touch setup.py
 
-A plugin module defines one or more hook implementation functions
-by decorating them with tox's ``hookimpl`` marker:
+contents of ``tox_fireworks.py``:
 
 .. code-block:: python
 
-    from tox import hookimpl
+    import pluggy
+
+    hookimpl = pluggy.HookimplMarker("tox")
 
     @hookimpl
     def tox_addoption(parser):
-        # add your own command line options
-
+        """Add command line option to display fireworks on request."""
 
     @hookimpl
     def tox_configure(config):
-        # post process tox configuration after cmdline/ini file have
-        # been parsed
+        """Post process config after parsing."""
+
+    @hookimpl
+    def tox_runenvreport(config):
+        """Display fireworks if all was fine and requested."""
+
+.. note::
 
-If you put this into a module and make it pypi-installable with the ``tox``
-entry point you'll get your code executed as part of a tox run.
+    See :ref:`toxHookSpecsApi` for details
+
+contents of ``setup.py``:
+
+.. code-block:: python
+
+    from setuptools import setup
 
+    setup(name='tox-fireworks', py_modules=['tox_fireworks'],
+          entry_points={'tox': ['fireworks = tox_fireworks']})
 
+Using the  **tox-** prefix in ``tox-fireworks`` is necessary for it to be an official plugin and
+makes finding it easy with e.g. ``pip search 'tox-'`` once it is released on PyPi.
+
+To make your new plugin discoverable by tox, you need to install it. During development you should
+install it with ``-e`` or ``--editable``, so that changes to the code are immediately active:
+
+.. code-block:: shell
+
+    $ pip install -e </path/to/tox-fireworks>
+
+
+Publish your plugin to PyPi
+---------------------------
+
+If you think the rest of the world could profit using your plugin you can publish it to PyPi.
+
+You need to add some more meta data to ``setup.py`` (see `cookiecutter-tox-plugin`_ for a complete
+example or consult the `setup.py docs <https://docs.python.org/3/distutils/setupscript.html>`_).
+
+..note::
+
+    Make sure your plugin project name is prefixed by `tox-` to be easy to find via e.g.
+    `pip serach tox-`
+
+You can and publish it like:
```

Comment:
this assumes your pip is already configured pypy remote, note? I would prefer here having here just a link with the more detailed and adequete https://packaging.python.org/tutorials/distributing-packages/#packaging-your-project

Reply:
I updated to the more specific link. I think though the basic example is a good start and the links to twine and packaging guide more guidance then,